### ISSUE TITLE
Set `--kubeconfig` flag as optional in `k3kcli cluster <cmd>` commands

### ISSUE THREAD

**Issue by enrichman** [rancher/k3k#153]
The `k3kcli cluster` commands need the `--kubeconfig` flag.

This should not be mandatory, and it should follow the same logic of `kubectl` ([see doc](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/#synopsis)):

> - If the --kubeconfig flag is set, then only that file is loaded. The flag may only be set once and no merging takes place.
>
> - If $KUBECONFIG environment variable is set, then it is used as a list of paths (normal path delimiting rules for your system). These paths are merged. When a value is modified, it is modified in the file that defines the stanza. When a value is created, it is created in the first file that exists. If no files in the chain exist, then it creates the last file in the list.
> 
> - Otherwise, ${HOME}/.kube/config is used and no merging takes place.

It should work by default, but if not I think we can avoid the merging of multiple files, if complicated, and just add the fallback to the default location `${HOME}/.kube/config`.

**Comment by VestigeJ** [comment]
@enrichman this is looking good! 

$ unset KUBECONFIG
$ echo $KUBECONFIG
$  ls .kube
```
cache  config
```

$  k3kcli cluster create --namespace default --servers 1 --agents 1 --mode virtual --persistence-type ephemeral defaultkubefolder
```
INFO[0000] Creating a new cluster [defaultkubefolder]
INFO[0000] Extracting Kubeconfig for [defaultkubefolder] cluster
INFO[0000] waiting for cluster to be available..
INFO[0065] certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1741045771: notBefore=2025-03-03 23:49:31 +0000 UTC notAfter=2026-03-03 23:50:34 +0000 UTC
INFO[0065] You can start using the cluster with:

		export KUBECONFIG=/home/ec2-user/defaultkubefolder-kubeconfig.yaml
		kubectl cluster-info
```



$ export KUBECONFIG=/home/ec2-user/rke2.yaml
$ k3kcli cluster create --namespace default --servers 1 --mode virtual --persistence-type ephemeral doozy
```
INFO[0000] Creating a new cluster [doozy]
INFO[0000] Extracting Kubeconfig for [doozy] cluster
INFO[0000] waiting for cluster to be available..
INFO[0075] certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1741037801: notBefore=2025-03-03 21:36:41 +0000 UTC notAfter=2026-03-03 21:37:49 +0000 UTC
INFO[0075] You can start using the cluster with:

		export KUBECONFIG=/home/ec2-user/doozy-kubeconfig.yaml
		kubectl cluster-info

```

$  k3kcli cluster create --kubeconfig=/etc/rancher/rke2/rke2.yaml --namespace default --servers 1 --agents 1 --mode virtual --persistence-type ephemeral zoort
```
INFO[0000] Creating a new cluster [zoort]
INFO[0000] Extracting Kubeconfig for [zoort] cluster
INFO[0000] waiting for cluster to be available..
INFO[0065] certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1741038381: notBefore=2025-03-03 21:46:21 +0000 UTC notAfter=2026-03-03 21:47:24 +0000 UTC
INFO[0065] You can start using the cluster with:

		export KUBECONFIG=/home/ec2-user/zoort-kubeconfig.yaml
		kubectl cluster-info

```

$ KUBECONFIG="/home/ec2-user/rke2.yaml"
$ k3kcli cluster create --namespace default --servers 1 --agents 1 --mode virtual --persistence-type ephemeral defaultconf
```
INFO[0000] Creating a new cluster [defaultconf]
INFO[0000] Extracting Kubeconfig for [defaultconf] cluster
INFO[0000] waiting for cluster to be available..
INFO[0070] certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1741043544: notBefore=2025-03-03 23:12:24 +0000 UTC notAfter=2026-03-03 23:13:32 +0000 UTC
INFO[0070] You can start using the cluster with:

		export KUBECONFIG=/home/ec2-user/defaultconf-kubeconfig.yaml
		kubectl cluster-info
```



$  k3kcli cluster create --kubeconfig=/etc/rancher/rke2/rke2.yaml --namespace default --servers 1 --agents 1 --mode virtual --persistence-type ephemeral dubles
```
INFO[0000] Creating a new cluster [dubles]
INFO[0000] Extracting Kubeconfig for [dubles] cluster
INFO[0000] waiting for cluster to be available..
INFO[0070] certificate CN=system:admin,O=system:masters signed by CN=k3s-client-ca@1741038228: notBefore=2025-03-03 21:43:48 +0000 UTC notAfter=2026-03-03 21:44:57 +0000 UTC
INFO[0070] You can start using the cluster with:

		export KUBECONFIG=/home/ec2-user/dubles-kubeconfig.yaml
		kubectl cluster-info
```

**Comment by enrichman** [comment]
@VestigeJ there is a small bug that I recently found though. Not critical, and also I'm not completely sure on how to fix it, so I guess it's ok to release this in the next 0.3.1, and just track it.

Basically kubectl supports also having multiple kubeconfig files in the same KUBECONFIG env var. This doesn't work for us.

**Comment by VestigeJ** [comment]
Yep I'm familiar with that from the cluster-api side of the k8s world. Are you trying to avoid using the existing tooling around contexts underneath those kubeconfig files?

**Comment by enrichman** [comment]
@VestigeJ not at all, but it looks like we used a "direct" approach, with the `[clientcmd.BuildConfigFromFlags](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#BuildConfigFromFlags)`.

I've finally found the clientcmd way to load the defaults, and eventually override the values with the `--kubeconfig` flag. I'll open a PR in a bit!
